app/store/ephemeral_store.ts

```diff
@@ -20,6 +26,31 @@ class EphemeralStore {
     private switchingToChannel = new Set<string>();
     private currentThreadId = '';
 
+    private loadingTeamChannels: {[serverUrl: string]: BehaviorSubject<number>} = {};
+
+    private getLoadingTeamChannelsSubject(serverUrl: string) {
+        if (!this.loadingTeamChannels[serverUrl]) {
+            this.loadingTeamChannels[serverUrl] = new BehaviorSubject(0);
+        }
+        return this.loadingTeamChannels[serverUrl];
+    }
+
+    constructor() {
+        DeviceEventEmitter.addListener(Events.FETCHING_TEAM_CHANNELS, (event) => this.onLoadingTeam(event));
```

Comment:
you can get rid of this event by  calling the `onLoadingTeam` from the action

Reply:
Also, this PR https://github.com/mattermost/mattermost-mobile/pull/6865 is doing something similar, I would love to have both things behaving in the same way